core/src/main/java/org/elasticsearch/index/query/FQueryFilterBuilder.java

```diff
@@ -89,9 +94,6 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     @Override
     public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
         // inner query builder can potentially be `null`, in that case we ignore it
-        if (this.queryBuilder == null) {
-            return null;
-        }
         Query innerQuery = this.queryBuilder.toQuery(parseContext);
```

Comment:
Actually this is not set through setter but through constructor, so it should always be okay.